src/ranged.cpp

```diff
@@ -2650,6 +2689,15 @@ target_handler::trajectory target_ui::run()
                 loop_exit_code = ExitCode::Timeout;
                 break;
             }
+
+            if( you->get_wielded_item()->has_flag( flag_RELOAD_AND_SHOOT ) ) {
+                if( !you->get_wielded_item()->ammo_remaining() && activity->reload_loc ) {
+                    you->mod_moves( -RAS_time );
```

Comment:
Don't mod moves, add moves to the activity.

Reply:
Ok, I'll change this.

Reply:
Sorry, can you please tell me how to change this? As stated above, PC will reload the bow in this part, by using mod_moves I mean the moves to reload is used up immediately here, and will not be counted any more in time_to_attack.

Reply:
```suggestion
                    activity->aif_duration += RAS_time;
```
Would this be a proper solution?

Reply:
@anothersimulacrum 

The way bow works before this PR:
1. Pressing f to choose the arrow and immediately reload with it.
2. Steady your aim or fire your bow, then the moves needed for reloading is consumed at once.
3. Not to shoot and quit the aim ui, with your bow unloaded immediately.

In step two, there were two bugs. The first one is, if you press f to enter the aim ui, then immediately quit it without any aiming action, you were supposed to get all the moves "refunded". The functions were wrote in that way, but failed to do so, thus you were losing moves by simply entering the aim ui. That's why auto re-aiming was banned for RAS weapon.
The second one is, if you aim at an adjacent tile and try pressing `.` to steady your aim, since that it can not be steadier, you just wait for one turn. Then, despite that there is an arrow notched, the moves needed for reloading is not consumed, firing your bow or quitting the aim ui will not consume the moves either.

This PR fixed those things, moves are not consumed if you don't try to shoot or aim your bow. If you try to make your aim steadier by pressing `.`, your will reload your bow first, then moves needed for reloading is consumed, and I believe that's the intended behavior.

https://github.com/CleverRaven/Cataclysm-DDA/blob/1aca3eb08b535090058514c8abf9396d91a4b591/src/activity_actor.cpp#L516-L536

https://github.com/CleverRaven/Cataclysm-DDA/blob/1aca3eb08b535090058514c8abf9396d91a4b591/src/ranged.cpp#L1060

The existing functions use mod_moves, so I just copied the method. I failed to find a way to "add moves to the activity", can you show me an example?

Reply:
Apologies, both for taking time to reply, and for the confusion - I thought it was more obvious (it's not!).
You add to `activity->moves_left` (and if you're feeling fancy, `moves_total`).

Reply:
Hi, thanks for your review. I asked @RenechCDDA for help, got the same advice, and unfortunately it didn't work.
If I understand it right, activity->moves_left is not for controlling moves needed when used in the aim activity.
https://github.com/CleverRaven/Cataclysm-DDA/blob/10a7e40a116696da62ace8fe8964621f8d88505b/src/activity_actor.cpp#L309-L314

https://github.com/CleverRaven/Cataclysm-DDA/blob/10a7e40a116696da62ace8fe8964621f8d88505b/src/activity_actor.cpp#L351-L357

https://github.com/CleverRaven/Cataclysm-DDA/blob/10a7e40a116696da62ace8fe8964621f8d88505b/src/ranged.cpp#L1516-L1537

In aim_activity_actor, moves_left is set to 1 when starting to aim, set to 0 when quitting the aim ui, and that's all. It is not modified or used elsewhere. Changing it from one  positive integer to another dosen't affect anything.

Instead, moves needed for aiming is consumed in do_aim, pressing `.` will consume 1 move for ten times, and choosing AIMED_SHOT etc will consume moves needed to reach certain aiming level.

Again, if I understand it right, using mod_moves is the only way here to affect moves that PC have.

Reply:
Also, this part seems to be wrong. PC's stamina is drained every time do_aim is called. I'm not sure, is it trying to compare 
moves_left with 1, but accidentally used moves PC has?
```
             // Only drain stamina on initial draw 
             if( you.get_moves() == 1 ) { 
                 mod_stamina_archery( you, relevant ); 
             } 
```

Reply:
Ack, you are right, my bad. I didn't realize the aiming activity was so funky and I gave you a bad suggestion. Use `mod_moves`.

Reply:
I don't see any reason why the stamina would be drained every call to do_aim, I don't have any explanation for why you'd see that, unfortunately.

Reply:
> I don't see any reason why the stamina would be drained every call to do_aim, I don't have any explanation for why you'd see that, unfortunately.

Well, I will deal with it in another PR. Thanks for your help!